services/video_indexer/actual_board_detection.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+
+def imageRowToBlackWhiteCountArray(image_row):
+    white_count = 0
+    black_count = 0
+
+    black_white_count_array = np.array([],dtype=int)
+
+    for i in image_row:
+        if i == 0:
+            if white_count != 0:
+                black_white_count_array = np.append(black_white_count_array, white_count)
+                white_count = 0
+            black_count += 1    
+        else:
+            if black_count != 0:
+                black_white_count_array = np.append(black_white_count_array,black_count)
+                black_count = 0
+            white_count += 1
+
+    if white_count != 0:
+        black_white_count_array = np.append(black_white_count_array, white_count)
+    else:
+        black_white_count_array = np.append(black_white_count_array, black_count)
+
+    return black_white_count_array
+
+def boardDetect(black_white_count_array, threshold):
+    same_count = 0
+    array_length = len(black_white_count_array)
+    current_base_value = black_white_count_array[0]
+
+    for i in range(0, array_length - 1):
```

Comment:
See video I linked in previous comment about smaller functions, as well as this one here by the same dude:
https://www.youtube.com/watch?v=EumXak7TyQ0
Again kinda subjective, but I really like his approach and we follow it on my current team in Ericsson.